server/model.go

```diff
@@ -10,13 +10,16 @@ import (
 	"net/http"
 	"os"
 	"path/filepath"
+	"sync"
 
 	"github.com/ollama/ollama/api"
 	"github.com/ollama/ollama/convert"
 	"github.com/ollama/ollama/llm"
 	"github.com/ollama/ollama/types/model"
 )
 
+var intermediateBlobs sync.Map
```

Comment:
To confirm, this is a map of blob digests to ggml model layer digests? Does intermediate mean f16 but not yet quantized?

Reply:
intermediate here means any blob that's not referenced in but used to produce something in the final manifest. here's a concrete example:

`FROM /path/to/safetensors/dir` creates a zip (intermediate) converts into a f16 (intermediate) quantizes into a q4_0 (final)

when you want to create another quantization, this map tracks the relationship between the zip and f16 so it's able to skip uploading the zip and reconverting the f16

Reply:
I think this probably deserves a comment describing its operation. I think most people aren't going to know what intermediate means here when they're going through the code.

I was trying to think os some other names, like maybe "blobCache" but then I think people will wonder why only some things are in the blobCache. Although why not shove everything into the blobCache? Why just put the "intermediate" stuff there?

Or the inverse; why not just put the intermediate stuff on desk and just use the normal machanism for pulling blobs?

Reply:
I'm also a little confused about why `sync.Map` here instead of `map[string]bool`. I don't _think_ there should be any contention here w/ mutexes given you're just shoving the same digest into the map.